### ISSUE TITLE
`open` command fails on valid TOML file

### ISSUE THREAD

**Issue by tylermumford** [nushell/nushell#7970]
### Describe the bug

I'm creating some TOML files for a personal project, and I've been previewing them with Nushell's `open` and `explore` commands, which are fantastic. But some files are valid TOML, yet Nushell cannot open them.

Here's a minimal TOML file that `open` cannot parse. I call it `nushell_toml_error.toml`.

```toml
command.build = "go build"

[command.deploy]
script = "./deploy.sh"
```

(I know I could rewrite this in a way that Nushell can parse. But I'm filing this because it's valid TOML, and it should be parsable and usable.)

### How to reproduce

1. Open a Nushell shell
2. Run `open nushell_toml_error.toml` (the snippet above)

When I run this, I get this output:
```
Error:
  × Error while parsing as toml
   ╭─[entry #94:1:1]
 1 │ open nushell_toml_error.toml
   ·      ───────────┬───────────
   ·                 ╰── Could not parse '/Users/tyler/dev/mmm/nushell_toml_error.toml' with `from toml`
   ╰────
  help: Check out `help from toml` or `help from` for more options or open raw data with `open --raw '/Users/tyler/dev/mmm/nushell_toml_error.toml'`

Error: nu::shell::cant_convert (link)

  × Can't convert to structured toml data.
   ╭─[entry #94:1:1]
 1 │ open nushell_toml_error.toml
   · ──┬─
   ·   ╰── can't convert string to structured toml data
   ╰────

```

### Expected behavior

I expected output like this:

```
╭─────────┬───────────────────────────────────────╮
│         │ ╭────────┬──────────────────────────╮ │
│ command │ │ build  │ go build                 │ │
│         │ │        │ ╭────────┬─────────────╮ │ │
│         │ │ deploy │ │ script │ ./deploy.sh │ │ │
│         │ │        │ ╰────────┴─────────────╯ │ │
│         │ ╰────────┴──────────────────────────╯ │
╰─────────┴───────────────────────────────────────╯
```

(This is the output I get if I parse the same file as above with Python 3.11's `tomllib`. I got that exact output via the nushell pipeline `python3.11 -c 'import tomllib; f = open("nushell_toml_error.toml", "rb"); t = tomllib.load(f); print(t)' | sd "'" '"' | from json`.)

### Configuration

| key                | value                                                             |
| ------------------ | ----------------------------------------------------------------- |
| version            | 0.75.0                                                            |
| branch             |                                                                   |
| commit_hash        |                                                                   |
| build_os           | macos-aarch64                                                     |
| build_target       | aarch64-apple-darwin                                              |
| rust_version       | rustc 1.67.0 (fc594f156 2023-01-24) (built from a source tarball) |
| cargo_version      | cargo 1.67.0 (8ecd4f20a 2023-01-10)                               |
| pkg_version        | 0.75.0                                                            |
| build_time         | 2023-01-31 20:00:59 +00:00                                        |
| build_rust_channel | release                                                           |
| features           | database, default, trash, which, zip                              |
| installed_plugins  |                                                                   |

### Additional context

I also validated the TOML snippet with [tomlv](https://github.com/BurntSushi/toml/tree/master/cmd/tomlv) and the Instaparse tool described in the formal [TOML grammar](https://github.com/toml-lang/toml/blob/1.0.0/toml.abnf). The `sd` in the pipeline above is [sd](https://github.com/chmln/sd), which I installed with Homebrew. I doubt it matters, but I'm using iTerm.

I understand if this is low-priority because the TOML snippet above can simply be rewritten. Let me know if this is a "if you want it fixed, please fix it yourself" type of situation.

Nushell is awesome. 💚

**Comment by sholderbach** [comment]
Thanks for the detailed report with a minimal reproducible example. That indeed looks like a bug and not like some inherent limitation between the data model of nushell and TOML (we had to restrict `to toml` as we were generating unrecoverable files)

**Comment by initinll** [comment]
Hello,
I am learning rust and this is my first OS contribution, raised one PR to fix this issue.

Fix summary - Upgraded toml crate version from 0.5.8 to 0.7.1.

**Comment by WindSoilder** [comment]
The issue is resolved by #7990
